state-chain/pallets/cf-swapping/src/tests/fees.rs

```diff
@@ -120,80 +111,220 @@ fn test_buy_back_flip() {
 			SwapQueue::<Test>::get(System::block_number() + u64::from(SWAP_DELAY_BLOCKS))
 				.first()
 				.expect("Should have scheduled a swap usdc -> flip"),
-			&Swap::new(1.into(), 1.into(), STABLE_ASSET, Asset::Flip, network_fee, None, [],)
+			&Swap::new(1.into(), 1.into(), STABLE_ASSET, Asset::Flip, NETWORK_FEE_AMOUNT, None, [],)
 		);
 	});
 }
 
 #[test]
-fn test_network_fee_calculation() {
-	fn take_fees_from_swap(
-		network_fee_percent: u32,
-		minimum_network_fee: AssetAmount,
-		chunk_amount: AssetAmount,
-		network_fee_collected: AssetAmount,
-		accumulated_stable_amount: AssetAmount,
-		is_internal: bool,
-	) -> (AssetAmount, AssetAmount) {
-		// Set the network fee and minimum network fee to the given values
-		if is_internal {
-			// Set the value of the normal network fee settings to 0 so we can test that the
-			// internal settings are used
-			NetworkFee::set(Permill::from_percent(0));
-			MinimumNetworkFee::<Test>::set(0);
-			InternalSwapNetworkFee::<Test>::set(Permill::from_percent(network_fee_percent));
-			InternalSwapMinimumNetworkFee::<Test>::set(minimum_network_fee);
-		} else {
-			NetworkFee::set(Permill::from_percent(network_fee_percent));
-			MinimumNetworkFee::<Test>::set(minimum_network_fee);
-			InternalSwapNetworkFee::<Test>::set(Permill::from_percent(0));
-			InternalSwapMinimumNetworkFee::<Test>::set(0);
-		}
+fn swap_uses_correct_network_fee() {
```

Comment:
I think it might be worth splitting this into individual tests (grouping them into a module would help keeping them organised). There is not that much boilerplate (and you could create a `setup_fees` if needed). Right now it is hard to link initiated swaps to the corresponding events: e.g. it is not clear which event corresponds to `IngressEgressFee`; you can sort of guess that it is the one where min network fee is not applied, but right now the same field is used to effectively identify a swap and aslo check its behaviour which isn't ideal.